pkg/sshd/sshd.go

```diff
@@ -178,35 +124,55 @@ func Configure(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrup
 	return cfg, nil
 }
 
-// FingerprintKey fingerprints a key and returns the colon-formatted version
-//
-// Params:
-// 	- key (ssh.PublicKey): The key to fingerprint.
-//
-// Returns:
-// 	- A string representation of the key fingerprint.
-func FingerprintKey(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
-	key := p.Get("key", nil).(ssh.PublicKey)
-	return Fingerprint(key), nil
+// GenSSHKeys generates the default set of SSH host keys.
+func GenSSHKeys(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+	log.Debugf(c, "Generating ssh keys for sshd")
+	// Generate a new key
+	out, err := exec.Command("ssh-keygen", "-A").CombinedOutput()
+	if err != nil {
+		log.Infof(c, "ssh-keygen: %s", out)
+		return nil, err
+	}
+	return nil, nil
 }
 
-// Fingerprint generates a colon-separated fingerprint string from a public key.
-func Fingerprint(key ssh.PublicKey) string {
-	hash := md5.Sum(key.Marshal())
-	buf := make([]byte, hex.EncodedLen(len(hash)))
-	hex.Encode(buf, hash[:])
-	// We need this in colon notation:
-	fp := make([]byte, len(buf)+15)
-
-	i, j := 0, 0
-	for ; i < len(buf); i++ {
-		if i > 0 && i%2 == 0 {
-			fp[j] = ':'
-			j++
-		}
-		fp[j] = buf[i]
-		j++
+func userInfoFromKey(key string) (*UserInfo, error) {
```

Comment:
Can you put this `func`, along with `controllerURLStr` into a `github.com/deis/pkg/controller` package?